### ISSUE TITLE
False positive `TC004` for optional type hints

### ISSUE THREAD

**Issue by sondrelg** [snok/flake8-type-checking#106]
Looks like 2.0.4 introduces false positive TC004 for code like this
```python
from typing import TYPE_CHECKING
        
if TYPE_CHECKING:
   from datetime import date
        
class Foo:
    def bar(self, *, baz: date | None):
        ...
```
